src/main/java/com/vmware/vipclient/i18n/base/cache/FormatCacheItem.java

```diff
@@ -4,28 +4,58 @@
  */
 package com.vmware.vipclient.i18n.base.cache;
 
-import java.util.HashMap;
-import java.util.Map;
+import com.vmware.vipclient.i18n.VIPCfg;
 
 public class FormatCacheItem implements CacheItem {
+
+	private String etag;
+	private long timestamp;
+	private Long maxAgeMillis = 86400000l;
+
 	public FormatCacheItem() {
-		
+
+	}
+
+	public String getEtag() {
+		return etag;
+	}
+
+	protected void setEtag(String etag) {
+		this.etag = etag;
+	}
+
+	public long getTimestamp() {
+		return timestamp;
+	}
+
+	protected void setTimestamp(long timestamp) {
+		this.timestamp = timestamp;
 	}
-	
-	public FormatCacheItem (Map<String, String> dataMap) {
-		super();
-		this.addCachedData(dataMap);
+
+	public Long getMaxAgeMillis() {
+		return maxAgeMillis;
 	}
-	
-	public final Map<String, String> cachedData = new HashMap<String, String>();
-	
-	public void addCachedData(Map<String, String> cachedData) {
-		if (cachedData != null)
-			this.cachedData.putAll(cachedData);
+
+	protected void setMaxAgeMillis(Long maxAgeMillis) {
+		this.maxAgeMillis = maxAgeMillis;
 	}
-		
-    public Map<String, String> getCachedData() {
-		return cachedData;
+
+	public boolean isExpired() {
+		// If offline mode only, cache never expires.
+		if (VIPCfg.getInstance().getVipServer() == null) {
+			return false;
+		}
+		// If maxAgeFromConfig is present, it means it is using the old way
+		// of caching expiration, so do not expire individual CacheItem object
+		if (VIPCfg.getInstance().getCacheExpiredTime() != 0) {
+			return false;
+		}
+
+		Long maxAgeResponse = this.getMaxAgeMillis();
+		if (maxAgeResponse != null) {
+			maxAgeMillis = maxAgeResponse;
+		}
+
+		return System.currentTimeMillis() - this.getTimestamp() >= maxAgeMillis;
```

Comment:
@Xiaochao8 , never put or invoke getter methods in synchronized blocks. This will affect performance. All threads must be able to get data simultaneously, and not have to wait in line.

The new setter methods accept the properties as set. They are also synchronized. This prevents threads from modifying individual properties at the same time which could lead to unmatched properties.

Reply:
Without synchronized keyword, how can you make sure the 2 variables are matched?

After this.getMaxAgeMillis() but before this.getTimestamp() , other thread may have changed timestamp, they are not matched in this case.

Reply:
@jessiejuachon I have tested with multiple threads, the scenario xiaochao mentioned can happen, so I add it to synchronized block.

You can also have a test.

Reply:
You are right! Thanks.